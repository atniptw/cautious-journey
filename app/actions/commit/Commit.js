// @flow
export const ADD_COMMIT = 'ADD_COMMIT';

export function addCommit(commit) {
  return { 
    ...commit, 
    type: ADD_COMMIT 
  };
}
