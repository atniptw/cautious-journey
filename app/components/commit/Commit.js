// @flow
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Commit.css';

import { Link } from 'react-router-dom';

const Commit = ({ hash, userName, date, files }) => (
      <div data-tid={hash}>
        <div>"Hash: {hash}"</div>
        <div>"User Name: {userName}"</div>
        <div>"Date: {date}"</div>
        <ul>
          {files.map((file, index) => (
            <li key={index}>{file}</li>
          ))}
        </ul>
      </div>
)

Commit.propTypes = {
  hash: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  files: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Commit
