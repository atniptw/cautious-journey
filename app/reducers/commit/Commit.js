// @flow
import { combineReducers } from 'redux';
import { ADD_COMMIT } from '../../actions/commit';

function commits(state: commit = {}, action) {
  switch (action.type) {
    case ADD_COMMIT:
      const id = action.hash;
      return {
        ...state,
        [action.hash]: {
          userName: action.userName,
          date: action.date,
          files: action.files
        }
      };
    default:
      return state;
  }
}

const codeAnalyzer = combineReducers({
  commits
})

export default codeAnalyzer;
