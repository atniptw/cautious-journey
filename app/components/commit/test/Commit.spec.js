import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Commit from '../Commit';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const commit = {
    hash: 'd715bad5-1c17-4821-bdc0-0154e9d06c7e',
    userName: 'Test User',
    date: '2018-02-14T21:00:42-05:00',
    files: ['example/path/file.txt', 'example/path/file2.txt']
  };
  return shallow(<Commit {...commit}/>);
}

describe('Commit component', () => {
  it('should should display commit', () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
