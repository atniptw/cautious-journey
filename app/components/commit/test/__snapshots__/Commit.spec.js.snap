// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit component should should display commit 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Commit
    date="2018-02-14T21:00:42-05:00"
    files={
      Array [
        "example/path/file.txt",
        "example/path/file2.txt",
      ]
    }
    hash="d715bad5-1c17-4821-bdc0-0154e9d06c7e"
    userName="Test User"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          "Hash: 
          d715bad5-1c17-4821-bdc0-0154e9d06c7e
          "
        </div>,
        <div>
          "User Name: 
          Test User
          "
        </div>,
        <div>
          "Date: 
          2018-02-14T21:00:42-05:00
          "
        </div>,
        <ul>
          <li>
            example/path/file.txt
          </li>
          <li>
            example/path/file2.txt
          </li>
        </ul>,
      ],
      "data-tid": "d715bad5-1c17-4821-bdc0-0154e9d06c7e",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "\\"Hash: ",
            "d715bad5-1c17-4821-bdc0-0154e9d06c7e",
            "\\"",
          ],
        },
        "ref": null,
        "rendered": Array [
          "\\"Hash: ",
          "d715bad5-1c17-4821-bdc0-0154e9d06c7e",
          "\\"",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "\\"User Name: ",
            "Test User",
            "\\"",
          ],
        },
        "ref": null,
        "rendered": Array [
          "\\"User Name: ",
          "Test User",
          "\\"",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "\\"Date: ",
            "2018-02-14T21:00:42-05:00",
            "\\"",
          ],
        },
        "ref": null,
        "rendered": Array [
          "\\"Date: ",
          "2018-02-14T21:00:42-05:00",
          "\\"",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              example/path/file.txt
            </li>,
            <li>
              example/path/file2.txt
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "example/path/file.txt",
            },
            "ref": null,
            "rendered": "example/path/file.txt",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "example/path/file2.txt",
            },
            "ref": null,
            "rendered": "example/path/file2.txt",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            "Hash: 
            d715bad5-1c17-4821-bdc0-0154e9d06c7e
            "
          </div>,
          <div>
            "User Name: 
            Test User
            "
          </div>,
          <div>
            "Date: 
            2018-02-14T21:00:42-05:00
            "
          </div>,
          <ul>
            <li>
              example/path/file.txt
            </li>
            <li>
              example/path/file2.txt
            </li>
          </ul>,
        ],
        "data-tid": "d715bad5-1c17-4821-bdc0-0154e9d06c7e",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"Hash: ",
              "d715bad5-1c17-4821-bdc0-0154e9d06c7e",
              "\\"",
            ],
          },
          "ref": null,
          "rendered": Array [
            "\\"Hash: ",
            "d715bad5-1c17-4821-bdc0-0154e9d06c7e",
            "\\"",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"User Name: ",
              "Test User",
              "\\"",
            ],
          },
          "ref": null,
          "rendered": Array [
            "\\"User Name: ",
            "Test User",
            "\\"",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"Date: ",
              "2018-02-14T21:00:42-05:00",
              "\\"",
            ],
          },
          "ref": null,
          "rendered": Array [
            "\\"Date: ",
            "2018-02-14T21:00:42-05:00",
            "\\"",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                example/path/file.txt
              </li>,
              <li>
                example/path/file2.txt
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "example/path/file.txt",
              },
              "ref": null,
              "rendered": "example/path/file.txt",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "example/path/file2.txt",
              },
              "ref": null,
              "rendered": "example/path/file2.txt",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
